src/main/java/com/intel/gkl/compression/IntelInflater.java

```diff
@@ -119,6 +127,9 @@ public void setInput(byte[] b, int off, int len) throws NullPointerException {
         if(b == null) {
             throw new NullPointerException("Input is null");
         }
+        if(off < 0 || off > b.length - len) {
+            throw new NullPointerException("Input offset is not valid.");
```

Comment:
This is not a Null Pointer exception.  See if you can find an Invalid Argument type of exception to throw.   This applies to all the the NullPointerException calls in this patch. Throw a descriptive error.

Reply:
Done !